lib/url.c

```diff
@@ -2519,6 +2519,20 @@ static CURLcode parse_proxy(struct Curl_easy *data,
   }
 
   /* now, clone the proxy host name */
+#ifdef USE_UNIX_SOCKETS
+  if(proxytype == CURLPROXY_SOCKS5_HOSTNAME
```

Comment:
Is there any reason to limit this to `SOCKS5_HOSTNAME` and not allow SOCKS4a too?

Reply:
I thought it may not work as dns resolution is done locally in SOCKS4a using the same function. But after quick test, looks like it should work. I will update for all socks proxies.

Reply:
Updated to support all socks proxies and not depend on the scheme prefix in proxy string. Please confirm if implementation/syntax is ok. I will then add a test.